File: C_Plus_Plus/251-500/p392.cpp
#include <string>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n = s.size(), m = t.size(), i=0, j=0;
        for (; i<n && j<m; j++) {
            if (s[i] == t[j]) i++;
        }
        
        return i == n;
    }
};

File: C_Sharp/251-500/p392.cs
namespace p392 {
    public class Solution {
        public bool IsSubsequence(string s, string t) {
            int n = s.Length, m = t.Length, i=0, j=0;
            for (; i<n && j<m; j++) {
                if (s[i] == t[j]) i++;
            }
            
            return i == n;
        }
    }
}